src/datasets/arrow_dataset.py

```diff
@@ -2960,7 +2960,7 @@ def _map_single(
         drop_last_batch: bool = False,
         remove_columns: Optional[List[str]] = None,
         keep_in_memory: bool = False,
-        load_from_cache_file: bool = None,
+        load_from_cache_file: Optional[bool] = None,
```

Comment:
`_map_single` expected a bool, so you can add this line at the beginning of the function:
```python
load_from_cache_file = load_from_cache_file if load_from_cache_file is not None else is_caching_enabled()
```
otherwise by default it would not return the cached data